_posts/2017-02-02-overlay-network-performance-testing.md

```diff
@@ -0,0 +1,75 @@
+---
+layout:     post
+title:      Overlay network performance testing
+date:       2017-02-02 10:00
+summary:    How do you decide on an overlay network to use for the next generation platform at SkyBet?
+category:   Infrastructure
+tags:       Networking, Performance, Containers, Docker, Kubernetes
+author:     martyn_ranyard
+---
+
+As one of the engineers working on our next-generation platform in the Bet tribe of **Sky Betting and Gaming**, I was tasked with choosing an overlay network for our container scheduler (we're going with Kubernetes but that's a story for a different post).  But just how do you choose an overlay network for a platform when you don't know exactly what will be running on it yet?  Here's how I went about it, some stats and some conclusions I came to.
+
+# What are the options?
+It turns out there are a wide variety of overlay networks to choose from and they all have different characteristics.  The current networking model for containers (e.g. Docker and Rkt - CoreOS' similar product) is called CNI and theoretically any overlay network that is CNI compliant can be used in Kubernetes.  The first wave of cutting down these networks is by ease of install - If I'm going to be testing networks en masse, I don't want to be spending days installing to find they're not suitable.
+
+This meant a much smaller list to benchmark - the ones that had kubeadm addons that can be installed quickly on a disposable cluster.  This list is :
+
+* Flannel - the default CoreOS (what we're probably going to use as our base OS) overlay network
+* Calico - the first major overlay network to have network policy support (ACLs for network, similar to firewalls)
+* Canal - This is effectively Flannel's overlay network with Calico's policy model, and is so similar to Flannel it's probably not worth testing
+* Weave - Another good alternative that has network policy support built in
+* Romana - This turned out to be not an overlay network per se, although it is a networking layer, but was later discarded because we want an overlay network
+
+# Requirements other than performance
+
+* **Sky Betting and Gaming** exists in a regulated business so we do require our new platform to have network policy support from the get go
+* We also have to be able to explain the stack to all our on-call engineers, so the simpler the better
+* Our newest datacentre is being set up with Cisco's ACI kit so support for that would be nice to have, but not in any way a necessity
+* The network must work in AWS as that is where our initial deployment will be (it can't interfere with AWS' networking)
+
+It turns out, other than the ACI nice to have, the three we whittled it down to all cover those if you add the Calico policy model to Flannel.
+
+# How do you test network performance
```

Comment:
If this is a question, it needs a question mark.  If this is a statement, it needs to lose the question tone.